manifests/tunnel.pp

```diff
@@ -3,31 +3,49 @@
   $source,
   $dest,
   $secret,
+  $user
 ) {
   ensure_resource('package', 'spiped', {'ensure' => 'present'})
   ensure_resource('file', '/etc/spiped', {'ensure' => 'directory'})
 
-  $unitfile = "/lib/systemd/system/spiped-${title}.service"
   $keyfile = "/etc/spiped/${title}.key"
 
+  if $::operatingsystem == 'Ubuntu' {
+    $unitfile = "/etc/init/spiped-${title}.conf"
+    $content = template('spiped/spiped.conf.erb')
+    $command = "restart spiped-${title}"
+  }
+  else {
+    $unitfile = "/lib/systemd/system/spiped-${title}.service"
+    $content = template('spiped/service.erb')
+    $command = "systemctl daemon-reload && systemctl restart spiped-${title}"
+  }
+
+  if $user {
+    $owner = $user
+  }
+  else {
+    $owner = root
+  }
```

Comment:
This is great, but I think could probably just add `$owner = 'root'` at the top as one of the arguments?